src/systems/eigen_system.C

```diff
@@ -43,14 +43,16 @@ EigenSystem::EigenSystem (EquationSystems & es,
                           const unsigned int number_in
                           ) :
   Parent           (es, name_in, number_in),
+  matrix_A         (nullptr),
+  matrix_B         (nullptr),
+  precond_matrix   (nullptr),
   eigen_solver     (EigenSolver<Number>::build(es.comm())),
   _n_converged_eigenpairs (0),
   _n_iterations           (0),
-  _is_generalized_eigenproblem (false),
-  _eigen_problem_type (NHEP),
   _use_shell_matrices (false),
   _use_shell_precond_matrix(false)
 {
+  set_eigenproblem_type(NHEP);
```

Comment:
Why can this no longer be set in the member initializer list?

Reply:
I thought this was a better way to do it because `_is_generalized_eigenproblem` depends on the `_eigen_problem_type`, therefore the two are always consistent if set using set_eigenproblem_type. 

Now that I think about it - it also makes sense to restrict set_eigenproblem_type with _can_add_matrices, because we can't change the type after matrices have been added.

Reply:
I can go back to the initialize list set, and I'm fine with that.. But I am going to put the latter in regardless.